plugins/modules/pacemaker_stonith.py

```diff
@@ -0,0 +1,228 @@
+
+
+class PacemakerStonith(StateModuleHelper):
+    module = dict(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=[
+                'present', 'absent', 'enabled', 'disabled']),
+            name=dict(type='str', required=True),
+            stonith_type=dict(type='str'),
+            stonith_options=dict(type='list', elements='str', default=list()),
+            stonith_operations=dict(type='list', elements='dict', default=list(), options=dict(
+                operation_action=dict(type='str'),
+                operation_options=dict(type='list', elements='str'),
+            )),
+            stonith_metas=dict(type='list', elements='str'),
+            stonith_argument=dict(type='dict', options=dict(
+                argument_action=dict(type='str', choices=['before', 'after', 'group']),
+                argument_options=dict(type='list', elements='str'),
+            )),
+            agent_validation=dict(type='bool', default=False),
+            wait=dict(type='int', default=300),
+        ),
+        required_if=[('state', 'present', ['stonith_type', 'stonith_options'])],
+        supports_check_mode=True
+    )
+
+    use_old_vardict = False
+    default_state = "present"
+
+    def __init_module__(self):
+        self.runner = pacemaker_runner(self.module, cli_action='stonith')
+        self.vars.set('previous_value', self._get())
+        self.vars.set('value', self.vars.previous_value, change=True, diff=True)
+
+    def _process_command_output(self, fail_on_err, ignore_err_msg=""):
+        def process(rc, out, err):
+            if fail_on_err and rc != 0 and err and ignore_err_msg not in err:
+                self.do_raise('pcs failed with error (rc={0}): {1}'.format(rc, err))
+            out = out.rstrip()
+            return None if out == "" else out
+        return process
+
+    def _get(self):
+        with self.runner('state name', output_process=self._process_command_output(False)) as ctx:
+            return ctx.run(state='status')
+
+    def _fmt_stonith_resource(self):
+        return dict([("resource_name", self.vars.stonith_type)])
+
+    # TODO: Pluralize operation_options in separate PR and remove this helper fmt function
+    def _fmt_stonith_operations(self):
+        modified_stonith_operations = []
+        for stonith_operation in self.vars.stonith_operations:
+            modified_stonith_operations.append(dict([("operation_action", stonith_operation.get('operation_action')),
+                                                    ("operation_option", stonith_operation.get('operation_options'))]))
+        return modified_stonith_operations
+
+    def state_absent(self):
+        with self.runner('state name', output_process=self._process_command_output(True, "does not exist"), check_mode_skip=True) as ctx:
+            ctx.run()
+            self.vars.set('value', self._get())
```

Comment:
You may use `__quit_module__()`, the counterpart of `__init_module__()` and do it only once.